zkevm-circuits/src/circuit_tools/memory.rs

```diff
@@ -237,6 +300,10 @@ impl<F: Field, C: CellType> MemoryBank<F, C> {
             require!(description, self.next => self.cur.expr() + condition.expr());
             // TODO(Brecht): add constraint that makes sure the table value remains the same when
             // not written
+            ifx!(not!(is_first_row) * not!(condition) => {
+                // TODO(Cecilia): Only works with Halo2 query API update
+                // require!(self.writes.0[0].column().expr() => self.writes.0[0].column().prev());
+            });
```

Comment:
Anything still to be done here? I haven't looked too closely yet if all constraints necessary are done here.

Reply:
In the ifx! I was trying constrain your todo: 
``` 
// TODO(Brecht): add constraint that makes sure the table value remains the same when
// not written
```
I was waiting for my Halo2 API updates but it can still be done with `a!`